### ISSUE TITLE
[Question] Who should NOT use fsrs4anki / the limits to generalization are in the training set

### ISSUE THREAD

**Issue by thiswillbeyourgithub** [open-spaced-repetition/fsrs4anki#631]
**Question**

Hi, I've been lurking around for months now about fsrs and something still not clear to me so I decided to come and ask. I do apologize in advance if those questions are irrelevant, I've been lurking for a while, waiting for someone to ask what I think is an obvious elephant in the room but nobody came so there's a decent chance I actually misunderstood something :/

My understanding is that FSRS is using machine learning to learn the free parameters in a formula of how retention can theoretically be modelled. And that the dataset for the actual learning is something the author gathered thanks to other users. But then comes the question of how representative this dataset is : garbage in means garbage out. So is it relevant to know if it was it lots of complex medical cards? quick vocabulary cards? from 10 users? 10 thousand users? Did they do breaks for week ends and vacation or did they have 10k day streaks? In my mind those question would have a very strong impact on the parameter space explored and so the applicability of FSRS to me.

Because the thing is I'm extremely dependant on Anki (more than 1k days streak as I'm both a medical student and do computer science etc) and I want to avoid jumping ship to FSRS just to learn to I was indeed an outlier and the learned parameters are infit for me. One of the weirder thing about my way to use anki is that I'm the dev behind [AnnA](https://github.com/thiswillbeyourgithub/AnnA_Anki_neuronal_Appendix/) (a python script that got out of hand and that I'll refactored when I finished med school) and use it daily : it basically adds a semantic loss to the review order, making me review less by reducing redundancies (more details: it just buries or create filtered decks based on `semantic similarity of the cards due today vs the cards you reviewed in the past few days + relative overdueness`).

**So to summarize here are my questions:**
1. What are the details behind the training dataset? What kind of user behave about like that training set (so would benefit greatly from FSRS) and what kind of user behave too differently from the training set hope being helped by FSRS?
2. In my specific case: do you have any specific guess on the probability that I would benefit a bit / a lot from FSRS? I'm really uncertain about the impact of using AnnA on FSRS

**Comment by aleksejrs** [comment]
https://github.com/open-spaced-repetition/fsrs4anki/blob/main/docs/tutorial.md#the-ultra-short-version

**Comment by L-M-Sherlock** [comment]
> 1. What are the details behind the training dataset?

Only the default weights are generated from the benchmark dataset. Here is the link: https://huggingface.co/datasets/open-spaced-repetition/FSRS-Anki-20k

It is random sampled from Anki users who have done 5k reviews. Due to the limit of privacy of Anki, we don't know what they learned. Here is the original discussion about the dataset: https://github.com/open-spaced-repetition/fsrs4anki/issues/493#issuecomment-1822429421

> 2\. do you have any specific guess on the probability that I would benefit a bit / a lot from FSRS?

I don't know. But you can check the accuracy of FSRS via the notebook-based optimizer: https://github.com/open-spaced-repetition/fsrs4anki/blob/main/fsrs4anki_optimizer.ipynb

It provides a lot of graphes and metrics to analyze FSRS in uploaded collections.
